### PR TITLE
[FLINK-26180] Update docs to introduce the compaction for FileSink.

### PR DESCRIPTION
**Pull request by pltbkd**
## What is the purpose of the change

This pull request update docs to introduce the compaction for FileSink.

## Brief change log

  - docs/connectors/datastream/filesystem is updated with a section introducing the compaction for FileSink.

### CODE REVIEW

File: docs/content/docs/connectors/datastream/filesystem.md [hunk 1]
```diff
@@ -958,6 +958,73 @@ val sink = FileSink
...
+When enabling compaction, you need to specify the {{< javadoc file="org/apache/flink/connector/file/sink/compactor/FileCompactStrategy.html" name="FileCompactStrategy">}}
+and the {{< javadoc file="org/apache/flink/connector/file/sink/compactor/FileCompactor.html" name="FileCompactor">}}.
+
+The {{< javadoc file="org/apache/flink/connector/file/sink/compactor/FileCompactStrategy.html" name="FileCompactStrategy">}} specifies
+when and which files get compacted. Currently, there are two parallel conditions: the target file size and the number of checkpoints get passed.
+Once the total size of the cached files has reached the size threshold or the number of checkpoints since the last compaction has reached the specified number, 
+the cached files will be scheduled to compact.
+
+The {{< javadoc file="org/apache/flink/connector/file/sink/compactor/FileCompactor.html" name="FileCompactor">}} specifies how to compact
+the give list of `Path` and write the result to {{< javadoc file="org/apache/flink/connector/file/sink/filesystem/CompactingFileWriter.html" name="CompactingFileWriter">}}. It could be classified into two types according to the type of the give `CompactingFileWriter`:
```
Comment (gaoyunhaii) [inline docs/content/docs/connectors/datastream/filesystem.md:1019]: The link to CompactingFileWriter seems not right. Also the same with the Chinese version

**Review by gaoyunhaii — COMMENTED** [review]
Thanks @pltbkd for the PR! I have one comment~

File: docs/content/docs/connectors/datastream/filesystem.md (hunk 1 above)
Comment (pltbkd) [reply docs/content/docs/connectors/datastream/filesystem.md]: Fixed.

**Comment by gaoyunhaii** [comment]
Thanks @pltbkd for the update! LGTM~ The failed test should be irrelevant~

**Pull request closed without merging**